forms/CheckboxField.php

```diff
@@ -6,59 +7,102 @@
  * @subpackage fields-basic
  */
 class CheckboxField extends FormField {
-
+	/**
+	 * @param int|bool $value
+	 *
+	 * @return $this
+	 */
 	public function setValue($value) {
-		$this->value = ($value) ? 1 : 0;
+		if($value) {
+			$this->value = 1;
+		} else {
+			$this->value = 0;
+		}
+
 		return $this;
 	}
 
+	/**
+	 * @return null|int
+	 */
 	public function dataValue() {
-		return ($this->value) ? 1 : NULL;
+		if($this->value) {
+			return 1;
+		}
+
+		return null;
 	}
 
+	/**
+	 * @return int
+	 */
 	public function Value() {
-		return ($this->value) ? 1 : 0;
+		if($this->value) {
+			return 1;
+		}
+
+		return 0;
 	}
 
+	/**
+	 * {@inheritdoc}
+	 */
 	public function getAttributes() {
-		$attrs = parent::getAttributes();
-		$attrs['value'] = 1;
+		$attributes = array(
+			'type' => 'checkbox',
+			'value' => 1,
+		);
+
+		if($this->Value()) {
+			$attributes['checked'] = 'checked';
```

Comment:
I just noticed; Prior to this change 'checked' would be null if not selected, but now it's not even assigned. Should we assign null in an else?